src/net/RTP/RTPPacket.cs

```diff
@@ -15,32 +15,127 @@
 //-----------------------------------------------------------------------------
 
 using System;
+using System.IO;
 
 namespace SIPSorcery.Net
 {
-    public class RTPPacket
+    public class RTPPacket : IDisposable
     {
-        public RTPHeader Header;
-        public byte[] Payload;
+        // Maximum size of an RTP packet
+        public const int RTP_PACKET_MAX_SIZE = 8192;
+
+        // Size of the extension header as defined by RFC 3550
+        public const int EXT_HEADER_SIZE = 4;
+
+        // Size of the fixed part of the RTP header as defined by RFC 3550
+        public const int FIXED_HEADER_SIZE = 12;
+
+        // MemoryStream storing the content of this packet
+        private MemoryStream _buffer;

```

Comment:
At first look, this can be replaced by https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream.

It would drastically reduce allocations and memory copy.